qiskit/circuit/commutation_checker.py

```diff
@@ -42,7 +63,13 @@ class CommutationChecker:
     evicting from the cache less useful entries, etc.
     """
 
-    def __init__(self, standard_gate_commutations: dict = None, cache_max_entries: int = 10**6):
+    def __init__(
+        self,
+        standard_gate_commutations: dict = None,
+        cache_max_entries: int = 10**6,
+        *,
+        gates: list[str] = None,
```

Comment:
You should make this a `set` so we have faster lookups.

Reply:
Done in https://github.com/Qiskit/qiskit/commit/351d5f11958e23d0b489b8138de0cb48a0e6a3c8